### PR TITLE
Remove IWithBoundedStash

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes
* Remove IWithBoundedStash

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
cc @ismaelhamed on this one - not sure if his stash changes would be adversely impacted by this.

File: src/core/Akka/Actor/Stash/IWithStash.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ namespace Akka.Actor
     /// that does not enforce any mailbox type see <see cref="IWithUnrestrictedStash"/>.
     /// </para>
     /// </summary>
-    public interface IWithStash : IWithUnrestrictedStash, IRequiresMessageQueue<IDequeBasedMessageQueueSemantics>
+    public interface IWithStash : IWithUnrestrictedStash, IRequiresMessageQueue<IBoundedDequeBasedMessageQueueSemantics>
     {
```
Comment (ismaelhamed) [inline src/core/Akka/Actor/Stash/IWithStash.cs:34]: Default deque must be unbounded, this is change in behavior. 

Keep in mind that when using these marker interfaces, the mailbox type will be resolved via the `akka.actor.mailbox.requirements` configuration in the hocon, not via the `StashFactory` (maybe that's the confusion).

Comment (Arkatufus) [reply src/core/Akka/Actor/Stash/IWithStash.cs:34]: `IWithStash` is the replacement for `IWithBoundedStash` interface, the unbounded behaviour is `IWithUnboundedStash`

Comment (Aaronontheweb) [reply src/core/Akka/Actor/Stash/IWithStash.cs:34]: Yeah this is an unsafe change.

Comment (Aaronontheweb) [reply src/core/Akka/Actor/Stash/IWithStash.cs:34]: Oh wait, are these the new APIs @ismaelhamed added?

Comment (Arkatufus) [reply src/core/Akka/Actor/Stash/IWithStash.cs:34]: Ah, I get it now... IWithStash is supposed to replace both `IWithBoundedStash` and `IWithUnboundedStash` and the behaviour is dictated by the settings?

File: src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveCore.Core.verified.txt [hunk 2]
```diff
@@ -1178,9 +1178,7 @@ namespace Akka.Actor
         void Become(Akka.Actor.UntypedReceive receive);
         void BecomeStacked(Akka.Actor.UntypedReceive receive);
     }
-    [System.ObsoleteAttribute("Use `IWithStash` with a configured BoundedDeque-based mailbox instead.")]
```
Comment (Arkatufus) [inline src/core/Akka.API.Tests/verify/CoreAPISpec.ApproveCore.Core.verified.txt:1181]: Obsolete since 1.5.0-alpha4

**Pull request closed without merging**